### PR TITLE
use correct `alg`

### PR DESCRIPTION
**Pull request by prbzrg**
The correct `alg` is decided in 
https://github.com/SciML/DiffEqBase.jl/blob/2dd4fafdf4857902a4401f11ae7e22934113a9f3/src/solve.jl#L1426-L1431
we should use it.

Fixes https://github.com/SciML/SciMLSensitivity.jl/issues/907

### CODE REVIEW

File: src/solve.jl [hunk 1]
```diff
@@ -1450,9 +1449,9 @@ function _solve_adjoint(prob, sensealg, u0, p, originator, args...; merge_callba
     end
 
     if isempty(args)
-        _concrete_solve_adjoint(_prob, nothing, sensealg, u0, p, originator; kwargs...)
+        _concrete_solve_adjoint(_prob, alg, sensealg, u0, p, originator; kwargs...)
     else
-        _concrete_solve_adjoint(_prob, args[1], sensealg, u0, p, originator,
+        _concrete_solve_adjoint(_prob, alg, sensealg, u0, p, originator,
```
Comment (ChrisRackauckas) [inline src/solve.jl:1454]: `alg` isn't always defined here? in the two branches above `alg` needs to be defined

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by prbzrg** [comment]
Why are we checking `isempty(args)` when `else` part is the same?
https://github.com/SciML/DiffEqBase.jl/blob/2dd4fafdf4857902a4401f11ae7e22934113a9f3/src/solve.jl#L1468-L1474

**Comment by ChrisRackauckas** [comment]
That should be splatting the rest of the args.

**Comment by prbzrg** [comment]
But `get_concrete_problem` doesn't take any more arguments:
https://github.com/SciML/DiffEqBase.jl/blob/2dd4fafdf4857902a4401f11ae7e22934113a9f3/src/solve.jl#L1114

**Comment by ChrisRackauckas** [comment]
Then we don't need that branch 😅

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by prbzrg** [comment]
CI failed, but it seems unrelated, and format check needs formatting for files that didn't change here.

**Comment by ChrisRackauckas** [comment]
It is unrelated, but I want to wait for @ErikQQY 's SDE fixes before merging more.

**Comment by ChrisRackauckas** [comment]
@ErikQQY see the failing remake tests here

**Pull request merged**